subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -82,6 +82,25 @@ Use other collections or I/O frameworks instead of `org.gradle.util.CollectionUt
 
 Gradle plugin authors may find the Designing Gradle Plugins subsection on link:https://guides.gradle.org/designing-gradle-plugins/#restricting_the_plugin_implementation_to_gradle_s_public_api[restricting the plugin implementation to Gradle's public API] helpful.
 
+[[sec:declaring_tasks]]
+=== Declaring tasks in a build script
+
+The task API gives a build author a lot of flexibility to declare tasks in a build script.
+For optimal readability and maintainability follow these rules:
+
+* The task type should be the only key-value pair that belongs within the parentheses after the task name.
+* Any other configuration logic for a task should be defined as part of api:org.gradle.api.Project#task(java.lang.String,groovy.lang.Closure)[] if possible.
+* <<sec:hello_world,Task actions>> should only be declared with the methods api:org.gradle.api.Task#doFirst(org.gradle.api.Action)[] or api:org.gradle.api.Task#doLast(org.gradle.api.Action)[].
```

Comment:
I appreciate that we link to something, but what this links to doesn't really make it clear what a task action is. I think this would be a wonderful opportunity to add to the glossary.

Reply:
I propose that we make the change with a new PR. Searching for "task action" in the user guide brings up 20 occurrences. It's great if we want to establish a glossary. However, I don't think it's necessary for this particular pull request.